Cython/Utility/CythonFunction.c

```diff
@@ -1018,7 +1018,11 @@ static PyTypeObject __pyx_CyFunctionType_type = {
     __pyx_CyFunction_methods,           /*tp_methods*/
     __pyx_CyFunction_members,           /*tp_members*/
     __pyx_CyFunction_getsets,           /*tp_getset*/
+#if PY_VERSION_HEX < 0x030900B1
     0,                                  /*tp_base*/
+#else
+    &PyCMethod_Type,                    /*tp_base*/
```

Comment:
This should be `PyCFunction_Type` right? Because the first struct element is `PyCFunctionObject`.

Both worked for me, but in an extreme edge case you could potentially trigger a method implemented from the base class, which may misuse `func_dict` (the one after `func`) because of its bigger size.

Reply:
The first field is `PyCMethodObject` in Py3.9+.

Reply:
Is that a Cython 3.0 change? I've only been running with tip of the 2.9 branch to get the fixes for 3.11, which didn't seem to be in the current 3.0 release.